Templates/JenkinsPipeline/README.md

```diff
@@ -0,0 +1,132 @@
+* the [basic pipeline](https://ibm.github.io/z-devops-acceleration-program/docs/branching-model-supporting-pipeline#the-basic-build-pipeline-for-main-epic-and-release-branches) when changes are merged into the branch `main` and
+* a [release pipeline](https://ibm.github.io/z-devops-acceleration-program/docs/branching-model-supporting-pipeline#the-release-pipeline-with-build-packaging-and-deploy-stages) to build and package the release candidate, that then can be deployed via UCD.
+
+Please review the pipeline stage `Pipeline setup` within the [template](Jenkinsfile#L165) that is configuring the subsequent stages including flags using as conditions when stages and jobs are executed. 
+
+Please make yourself familiar with the [Git branching for mainframe development](https://ibm.github.io/z-devops-acceleration-program/docs/git-branching-model-for-mainframe-dev/#characteristics-of-mainline-based-development-with-feature-branches) documentation, that this template is implementing.
+
+### Pipeline variables
+
+In a default setup, the feature and basic pipelines are triggered for each new commit as soon as the Multibranch integration detects the changes.
+
+The pipeline allows to override the values of the below variables when manually requesting the pipeline. This is especially useful when the application team want to create a release candidate package for higher test environments and production.
+
+Parameter | Description
+--- | ---
+pipelineType     | Pipeline type - either build, release or preview. (Default: build)
+verbose          | Boolean flag to control logging of build framework. (Default: false)
+<!-- releaseType      | Release type - major, minor, patch as input to compute the release version and to set the release candidate and release git tags. (Default: patch) //-->
+
+### Feature Branch pipeline
+
+The pipeline for feature branches executes the following steps:
+
+* Setup
+* Clone
+* Build
+* SonarQube Analysis
+* Create UCD Component version
+
+This pipeline can be manually requested to set the *pipelineType* variable `preview` to run zAppBuild in preview mode.
+
+Overview of the pipeline:  
+
+![Jenkins Feature Branch Pipeline](images/jenkins-pipeline-feature-branch.png)
+
+Pipelines for Feature, Epic and Release Maintenance branches perform the same steps like a feature branch pipeline. 
+
+### Basic build pipeline when merging into Main
+
+The basic build pipeline for the main branch contains the following stages:
+
+* Setup
+* Clone
+* Build (with TEST options)
+* SonarQube Analysis
+* Create UCD Component version
+* Request UCD Deployment to the integration test environment
+
+It run automatically when there is a new commit to a repository. The common backend script `dbbBuild.sh` will automatically extract the baseline git tag based on the information in the [baselineReference.config](../Common-Backend-Scripts/samples/baselineReference.config) file that is expected to be maintained in the application's git repository.
```

Comment:
It runs automatically when there is a new commit to a repository. The `dbbBuild.sh` script (from the Common Backend Scripts) will automatically extract the Git tag based on the information in the [baselineReference.config](../Common-Backend-Scripts/samples/baselineReference.config) file. The `baselineReference.config` file is expected to be maintained in the application's Git repository.